pkg/util/kubernetes/apiserver/resourcetypes_test.go

```diff
@@ -0,0 +1,258 @@
+	resetCache()
+
+	client := fakeclientset.NewClientset()
+	fakeDiscoveryClient := client.Discovery().(*fakediscovery.FakeDiscovery)
+	fakeDiscoveryClient.Resources = []*v1.APIResourceList{
+		{
+			GroupVersion: "v1",
+			APIResources: []v1.APIResource{
+				{Kind: "Pod", Name: "pods"},
+			},
+		},
+	}
+
+	err := InitializeGlobalResourceTypeCache(fakeDiscoveryClient)
+	assert.NoError(t, err)
+
+	tests := []struct {
+		name    string
+		kind    string
+		group   string
+		want    string
+		wantErr bool
+	}{
+		{
+			name:    "Cache hit for Pod/v1",
+			kind:    "Pod",
+			group:   "v1",
+			want:    "pods",
+			wantErr: false,
+		},
+		{
+			name:    "Cache miss for unknown kind",
+			kind:    "UnknownKind",
+			group:   "unknownGroup",
+			want:    "",
+			wantErr: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := GetResourceType(tt.kind, tt.group)
+			if tt.wantErr {
+				assert.Error(t, err)
+			} else {
+				assert.NoError(t, err)
+				assert.Equal(t, tt.want, got)
+			}
+		})
+	}
+}
+
+func TestDiscoverResourceType(t *testing.T) {
+	resetCache()
+
+	client := fakeclientset.NewClientset()
+	fakeDiscoveryClient := client.Discovery().(*fakediscovery.FakeDiscovery)
+	fakeDiscoveryClient.Resources = []*v1.APIResourceList{
+		{
+			GroupVersion: "apps/v1",
+			APIResources: []v1.APIResource{
+				{Kind: "Deployment", Name: "deployments"},
+			},
+		},
+	}
+
+	err := InitializeGlobalResourceTypeCache(fakeDiscoveryClient)
+	assert.NoError(t, err)
+
+	tests := []struct {
+		name    string
+		kind    string
+		group   string
+		want    string
+		wantErr bool
+	}{
+		{
+			name:    "Find Deployment in apps/v1",
+			kind:    "Deployment",
+			group:   "apps",
+			want:    "deployments",
+			wantErr: false,
+		},
+		{
+			name:    "Resource not found",
+			kind:    "UnknownKind",
+			group:   "unknownGroup",
+			want:    "",
+			wantErr: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := resourceCache.discoverResourceType(tt.kind, tt.group)
+			if tt.wantErr {
+				assert.Error(t, err)
+			} else {
+				assert.NoError(t, err)
+				assert.Equal(t, tt.want, got)
+			}
+		})
+	}
+}
+
+func TestPrepopulateCache(t *testing.T) {
+	resetCache()
+
+	client := fakeclientset.NewClientset()
+	fakeDiscoveryClient := client.Discovery().(*fakediscovery.FakeDiscovery)
+	fakeDiscoveryClient.Resources = []*v1.APIResourceList{
+		{
+			GroupVersion: "v1",
+			APIResources: []v1.APIResource{
+				{Kind: "Pod", Name: "pods"},
+				{Kind: "Secret", Name: "secrets"},
+				{Kind: "ConfigMap", Name: "configmaps/status"},
+			},
+		},
+	}
+
+	resourceCache = &ResourceTypeCache{
+		kindGroupToType: make(map[string]string),
+		discoveryClient: fakeDiscoveryClient,
+	}
+
+	tests := []struct {
+		name    string
+		wantErr bool
+	}{
+		{
+			name:    "Prepopulate cache successfully",
+			wantErr: false,
+		},
+	}
+
+	for _, tt := range tests {
```

Comment:
Since there is only one test here, I think you can simplify it by removing this loop